### PR TITLE
Np 46206 list country codes

### PR DESCRIPTION
**Pull request by torarnet**

### CODE REVIEW

File: cristin-person/src/main/java/no/unit/nva/cristin/person/country/ListCountriesApiClient.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        super(httpClient);
+    }
+
+    public Countries executeRequest() throws ApiGatewayException {
+        Map<String, String> params = emptyMap();
+        var queryUri = createCristinQueryUri(params, CRISTIN_COUNTRIES_PATH);
+        var start = System.currentTimeMillis();
+        var response = queryUpstream(queryUri);
+        var countries = getCountries(response);
+        var totalProcessingTime = calculateProcessingTime(start, System.currentTimeMillis());
```
Comment (torarnet) [inline cristin-person/src/main/java/no/unit/nva/cristin/person/country/ListCountriesApiClient.java:42]: im considering removing processingtime as I am caching the lambda response with cloudfront

Comment (brinxmat) [reply cristin-person/src/main/java/no/unit/nva/cristin/person/country/ListCountriesApiClient.java:42]: Maybe only interesting for debug, so probably — we have other ways of getting this info now and as you point out the cache makes the info stale

File: cristin-person/src/main/java/no/unit/nva/cristin/person/country/ListCountriesHandler.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import nva.commons.apigateway.RequestInfo;
+import nva.commons.apigateway.exceptions.ApiGatewayException;
+import nva.commons.core.Environment;
+
+@SuppressWarnings("unused")
+public class ListCountriesHandler extends ApiGatewayHandler<Void, Countries> {
+
+    private final ListCountriesApiClient apiClient;
+
+    public ListCountriesHandler() {
```
Comment (brinxmat) [inline cristin-person/src/main/java/no/unit/nva/cristin/person/country/ListCountriesHandler.java:19]: We never return a single country value, only a list?

File: cristin-person/src/main/java/no/unit/nva/cristin/person/model/cristin/CristinCountry.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+import static no.unit.nva.cristin.model.JsonPropertyNames.CODE;
+import static no.unit.nva.cristin.model.JsonPropertyNames.CODE_ALPHA_3;
+import static no.unit.nva.cristin.model.JsonPropertyNames.NAME;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Map;
+import no.unit.nva.commons.json.JsonSerializable;
+import no.unit.nva.cristin.person.model.nva.Country;
+
+public record CristinCountry(@JsonProperty(CODE) String code,
```
Comment (brinxmat) [inline cristin-person/src/main/java/no/unit/nva/cristin/person/model/cristin/CristinCountry.java:11]: In many ways, I'd prefer an id URI on each country that could be referenced and de-referenced

Comment (brinxmat) [reply cristin-person/src/main/java/no/unit/nva/cristin/person/model/cristin/CristinCountry.java:11]: Possibly a new PR

File: cristin-person/src/main/java/no/unit/nva/cristin/person/country/ListCountriesHandler.java (hunk 2 above)
Comment (torarnet) [reply cristin-person/src/main/java/no/unit/nva/cristin/person/country/ListCountriesHandler.java:19]: yes, we only list the whole dataset. it is only 255 results and i see no reason to do a lookup. it is also not updated often. if we want to have a lookup i have to implement in cristin api as well.

File: cristin-person/src/main/java/no/unit/nva/cristin/person/model/cristin/CristinCountry.java (hunk 3 above)
Comment (torarnet) [reply cristin-person/src/main/java/no/unit/nva/cristin/person/model/cristin/CristinCountry.java:11]: do you have an example uri we should use?

File: cristin-person/src/main/java/no/unit/nva/cristin/person/country/ListCountriesApiClient.java (hunk 1 above)
Comment (torarnet) [reply cristin-person/src/main/java/no/unit/nva/cristin/person/country/ListCountriesApiClient.java:42]: i will remove this field

File: cristin-person/src/main/java/no/unit/nva/cristin/person/country/ListCountriesHandler.java (hunk 2 above)
Comment (torarnet) [reply cristin-person/src/main/java/no/unit/nva/cristin/person/country/ListCountriesHandler.java:19]: if you want a lookup i can add a lookup in a new PR?

**Review by brinxmat — APPROVED** [review]

File: cristin-person/src/main/java/no/unit/nva/cristin/person/model/cristin/CristinCountry.java (hunk 3 above)
Comment (brinxmat) [reply cristin-person/src/main/java/no/unit/nva/cristin/person/model/cristin/CristinCountry.java:11]: probably, the current /country/ + unique code (i.e. the ISO-code)

Comment (torarnet) [reply cristin-person/src/main/java/no/unit/nva/cristin/person/model/cristin/CristinCountry.java:11]: iso 2 or 3 letter alpha3? we have both

**Pull request merged**